File: Lesson 1 - Iterations/Binary_Gap.py
def solution(N):
    # write your code in Python 3.6
    # in binary
    binary_arg = format(N,"b")

    splitted_binary = [int(char) for char in binary_arg]

    # collect 0's for each squence
    # each count to be collected in here
    counts=[]
    # count_max = float('inf')

    counter=0
    for char in range(len(splitted_binary)):
        
        if (char == len(splitted_binary)-1) and (splitted_binary[char] == 0):
            counter += 1
            counts.append(counter)

        elif (splitted_binary[char]) == 0:
            counter += 1

        elif (splitted_binary[char]) == 1:
            counts.append(counter)
            counter=0


    return max(counts)

print(solution(200))

File: Lesson 1 - Iterations/Binary_Gap.js
function dec2bin(dec) {
    return (dec >>> 0).toString(2);
  }

function solution(N) {
    let splittedBinary = []
    // converting decimal into binary format
    let nBinary = dec2bin(N)

    // iterating over binary for splitting into a list obj
    for (let i = 0; i < nBinary.length; i++) {
        splittedBinary.push(nBinary.charAt(i))
    }       
    
    let counts=[]
    let counter=0
    // iterating over list for counting gaps
    for (let i=0; i < splittedBinary.length; i++) {
        // checking if element is 0 and i is the last element
        if (i == (splittedBinary.length-1) && splittedBinary[i] == 0) {
            
            counter += 1
            counts.push(counter)
        }
        // if element zero increase by one
        else if (splittedBinary[i] == 0) {
            
            counter += 1
        }

        // if element one push the counter value to the list and reset counter
        else if (splittedBinary[i] == 1) {
            
            counts.push(counter)
            counter = 0
        }
    }

    return Math.max.apply(Math,counts)
}

console.log(solution(200));